### DISCUSSION TITLE
Is it possible to use Three.js scene as input

### DISCUSSION THREAD

**Discussion by mpena211** [google/model-viewer#3294] [Ideas]
HI, first off I love this library. Extremely easy to set up and use. 

I get that model-viewer implies just viewing and interacting with the model, and it only supports .gltf/.glb file types. However I was wondering if it is possible to *theoretically* use a three.js scene. I had a quick look through of the source code and saw that you are using three.js for at least part of the pipeline to load the models. 

What I had in mind, was to essentially make a three.js scene, with a character inside of it (maybe a model), and then walk around inside the scene. I think it is possible to do this in vanilla three.js but, I really enjoy using your library to handle the loading & displaying of models, and do not want to rewrite that functionality if possible. Thanks!

**Comment by elalish** [comment]
If you scene can be saved as a glTF, this will work simply by making your own camera controls, which can be done externally to MV in your own JS. If you need all the bells and whistles of three's API, then I'd say you can fork us or use `getOwnSymbols` to grab our internal `$scene` and do what you want. However, you're outside our public API, so you'll be on your own, especially regarding breaking changes.

**Reply by mpena211** [reply]
Oh that is awesome, would much rather make a custom camera. Once I write the camera-controls, I can just pass it into the parameter on the model viewer object? Also do you know if someone has done something similar to this already to give me a head start? Thanks!

**Reply by elalish** [reply]
Start with this: https://modelviewer.dev/examples/stagingandcameras/#turnSkybox

You'll want to adjust `camera-orbit` and `camera-target` instead. Maybe set radius (and min radius) to zero. Don't include the `camera-controls` attribute so that our controls will be disabled.

**Reply by elalish** [reply]
Would love to see what you create! Feel free to share in the show&tell section, or mention us on Twitter.

**Reply by allenwixted** [reply]
On a similar note, are there any examples for passing a scene from three.js to MV? Trying to make a sofa builder and pass to AR

**Reply by elalish** [reply]
Here's one: https://github.com/mrdoob/3d-file-viewer/blob/master/script.js

**Reply by Zaniyar** [reply]
Is this also possible to do with the AR scene? I know you could do this with webXR, but since Apple is doing Apple things and it's not working on iOS but the model-viewer AR is working on iOS and Android I would want to know if we can have an interactive AR scene with the model-viewer?

**Reply by elalish** [reply]
No, there's no customization available for QuickLook - please ask Apple (@grorg) for WebXR support so this can be possible.
